[FEAT/#208] OtpScreen UI 구현

angryPodo opened the pull request:
## Related issue 🛠
- closed #208 

## Work Description ✏️
- UI 스펙에 맞춰 화면 구현
- 기본적인 네비게이션 세팅

## Screenshot 📸

https://github.com/user-attachments/assets/6fc1c4e1-ba21-46bc-832a-feaecaefb048

## To Reviewers 📢
#206 PR을 받아와서 사용했습니다. 근데 이번 PR Changed가 100+ 정도 밖에 안되어서..ㅎㅎ 양해부탁드립니다.
#206 이 머지된 후에 리뷰하셔도 좋습니다

angryPodo committed:
[ADD/#205] Create OTP feature module

- Add `presentation:otp` module to project.
- Include new module in `settings.gradle.kts`.
- Add `.gitignore` to `presentation/otp`.
- Create `AndroidManifest.xml` for `presentation:otp`.
- Add `espressoCore` dependency.

angryPodo committed:
[FEAT/#205] OtpTextField component

Implement OtpTextField composable for OTP input.
- Displays 6 input boxes for OTP digits.
- Handles input validation (digits only, max 6 characters).
- Changes appearance based on input state (empty, filled, error).
- Includes preview for different states.

angryPodo committed:
[mod/#205] Implement OtpTextField component

This commit introduces the `OtpTextField` composable, a custom UI element for OTP (One-Time Password) input.

- **`OtpTextField`**: The main composable function that handles OTP input and display.
    - Takes `otpText`, `onOtpTextChange`, and `isError` as parameters.
    - Uses `BasicTextField` for input and displays OTP characters in individual boxes.
- **`OtpChar`**: A private composable function responsible for rendering each character box within the `OtpTextField`.
    - Styles the box based on whether it's filled or in an error state.
    - Updates border color, background color, and text color accordingly.
- **Preview**: Includes a preview function (`OtpTextFieldPreview`) to demonstrate different states of the `OtpTextField` (empty, filled, error, full error).

angryPodo committed:
[MOD/#205] OtpTextField: Update otpText to lambda and handle keyboard done action

- Changed the `otpText` parameter in `OtpTextField` from a `String` to a `() -> String` lambda to align with Jetpack Compose best practices.
- Implemented `KeyboardActions` to clear focus when the "Done" action is triggered on the keyboard.
- Updated `KeyboardOptions` to specify `ImeAction.Done`.

angryPodo committed:
[FEAT/#205] Otp Screen and Navigation Setting

Commit:
[CI] Apply Spotless formatting

angryPodo committed:
[MOD/#205] Lint format

angryPodo committed:
Merge remote-tracking branch 'origin/feat/#205-otp-textfield' into feat/#205-otp-textfield

angryPodo committed:
Merge remote-tracking branch 'origin/feat/#205-otp-textfield' into feat/#208-otp-screen-ui

angryPodo committed:
[FEAT/#208] Add OtpScreen UI

This commit introduces the UI for the OTP (One-Time Password) screen.

Key changes:
- Added `OtpRoute` composable function to handle navigation.
- Implemented `OtpScreen` composable function which includes:
    - A `BackTopAppBar` for navigation.
    - An `OtpTextField` for OTP input.
    - Informational text about the OTP.
    - A clickable text for users who haven't received an OTP.
    - A "인증하기" (Authenticate) button.
- Added a preview function `OtpScreenPreview`.

angryPodo committed:
[MOD/#208] Modify OtpRoute navigation

nhyeonii commented on gradle/libs.versions.toml:
```diff
@@ -155,6 +156,7 @@ room-compiler = { group = "androidx.room", name = "room-compiler", version.ref =
 # ML Kit
 mlkit-text-recognition = { module = "com.google.mlkit:text-recognition", version.ref = "mlkitTextRecognition" }
 kotlinx-coroutines-play-services = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-play-services", version.ref = "coroutinesPlayServices" }
+androidx-espresso-core = { group = "androidx.test.espresso", name = "espresso-core", version.ref = "espressoCore" }
```
p3 : 전 PR에서 기존 View 기반 라이브러리를 Compose에서 억지로 래핑하거나 함께 써야 하는 경우를 대비해 Espresso를 의존성에 추가히셨다가 삭제하셨던 걸로 기억합니다. 이번 PR에서 다시 추가하신 특별한 이유가 있을까요 ??

angryPodo replied:
? 모듈만들면서 생겼나보다 삭제할게요

nhyeonii reviewed (approved):
확인 완료!! 고생하셨습니닷

angryPodo committed:
[DEL/#208] 에스프레소 의존성 삭제

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+}
+
+@Composable
+fun OtpScreen(
```
private?

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+    onTextClick: () -> Unit,
+    onButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    isButtonEnable: () -> Boolean
```
P1: 필수인자가 위로!!

Hyobeen-Park replied:
그리고 screen에서 modifier를 받아오는 이유가 무엇인가요?

angryPodo replied:
어쨌든 컴포넌트로서 사용하니까 슬롯을 미리 뚫어놨습니다

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
P1: 여기서 statusBar 색상 지정 안해주면 하이링오렌지일 것 같은데 확인 한 번만 부탁드려요!!

angryPodo replied:
이거는 나중에 한번에 할게요. 36에서는 아에 Deprecated되어서

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+            text = "인증 번호가 오지 않았나요?",
+            style = HilingualTheme.typography.bodyM14.copy(
+                textDecoration = TextDecoration.Underline
+            ),
```
```suggestion
            style = HilingualTheme.typography.bodyM14,
            textDecoration = TextDecoration.Underline
```
P2: 이렇게 하는건 어때요

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+            color = HilingualTheme.colors.gray500,
+            modifier = Modifier
+                .padding(horizontal = 16.dp)
+                .fillMaxWidth()
```
P2: fillMaxWidth는 왜 하신건가요?

angryPodo replied:
빼보실??

Hyobeen-Park replied:
안 하면 중앙정렬 돼서 넣으신거 맞죠?? 근데 `fillMaxWidth()` 사용하시면 터치 영역이 텍스트 영역이 아닌 가로 전체로 잡혀요! 그래도 괜찮다면 이대로 진행해도 좋지만 피그마대로 구현하고 싶다면 `Modifier.align()` 속성 사용해보시면 좋을 듯 합니다😊

HAJIEUN02 replied:
하 너무 웃기다 민재오빠 코리 구경하는 재미가 있네요
빼보실??

Hyobeen-Park replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+
+@Preview
+@Composable
+fun OtpScreenPreview() {
```
P1: private해줘잉

Hyobeen-Park commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -0,0 +1,152 @@
+            onOtpTextChange = { value = it }
+        )
+
+        Spacer(Modifier.height(40.dp))
```
P1: 에러 텍스트는 Otp텍필 안에서 보여줄건가요?? 그렇다면 여기 spacer는 13.dp로 해야 할 듯 합니다 그리고 추가로 에러 텍스트 영역도 넣어줘야 해요!

angryPodo replied:
그거 PR분리하려고 안해놨습니다ㅎㅎ

Hyobeen-Park reviewed (commented):
캬~ 빠르다 빨라

angryPodo committed:
[MOD/#208] Make OtpScreen private and reorder parameters

The `OtpScreen` composable function is now private.
The `isButtonEnable` parameter has been moved to be the last function parameter.
The `Text` composable for "인증 번호가 오지 않았나요?" now directly applies `textDecoration`.

angryPodo committed:
[MOD/#208] Make OtpScreen private and reorder parameters

The `OtpScreen` composable function is now private.
The `isButtonEnable` parameter has been moved to be the last function parameter.
The `Text` composable for "인증 번호가 오지 않았나요?" now directly applies `textDecoration`.

angryPodo committed:
Merge remote-tracking branch 'origin/develop' into feat/#208-otp-screen-ui

# Conflicts:
#	presentation/main/src/main/java/com/hilingual/presentation/main/MainScreen.kt
#	presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt
#	presentation/otp/src/main/java/com/hilingual/presentation/otp/navigation/OtpNavigation.kt

Hyobeen-Park reviewed (approved):
아 맞다 어푸푸!!ㅎㅎㅎ 텍스트 fillMaxWidth 이슈만 다시 한 번 확인 부탁드려요😊

Daljyeong commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -29,33 +33,119 @@ import androidx.compose.runtime.setValue
 
 @Composable
-fun OtpScreen(
+fun OtpRoute(
```
p3: 여기에 `internal`이 없는 이유가 궁금합니당

angryPodo replied:
요거는 그냥 안넣었어요ㅋㅋ 다음 Pr에서 넣겠음

Daljyeong reviewed (approved):
빠르다 빨라🫢 에러 텍스트도 기대되네욤 수고하셨습니당!!

nahy-512 commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -29,33 +33,119 @@ import androidx.compose.runtime.setValue
     modifier: Modifier = Modifier
 ) {
     val focusManager = LocalFocusManager.current
+    var value by remember { mutableStateOf("") }
```
p3
저는 이 value라는 네이밍이 조금 모호할 수 있다는 생각이 드는데, 혹시 어떻게 생각하시나요?
기존 `otpText`란 명칭에서 변경하신 이유가 있는지 궁금해요!

nahy-512 replied:
아 다음 작업에서 `otpCode`로 바꿨구나! 화긴

angryPodo replied:
어차피 이런 단계에서 작업한거 뷰모델로 다 State넘기니까 신경 안써도 댑니다

nahy-512 reviewed (approved):
확인 완료!! 고생하셨습니다~!!!

HAJIEUN02 commented on presentation/otp/src/main/java/com/hilingual/presentation/otp/OtpScreen.kt:
```diff
@@ -29,33 +33,119 @@ import androidx.compose.runtime.setValue
+
+@Preview
+@Composable
+private fun OtpScreenPreview() {
```
p5: 왜 Preview로 영상 안 찍고 빌드해서 돌리시는지 궁금해염
